debezium-core/src/main/java/io/debezium/heartbeat/HeartbeatImpl.java

```diff
@@ -77,6 +77,11 @@ public void heartbeat(Map<String, ?> partition, Map<String, ?> offset, BlockingC
     public void forcedBeat(Map<String, ?> partition, Map<String, ?> offset, BlockingConsumer<SourceRecord> consumer)
             throws InterruptedException {
         LOGGER.debug("Generating heartbeat event");
+        final SourceRecord record = heartbeatRecord(partition, offset);
+        if (record.sourceOffset() == null || record.sourceOffset().isEmpty()) {
```

Comment:
@jpechane Couldn't this check be rather applied on the passed `offset` parameter? Avoids the creation of the record just for this.